### PR TITLE
Build: Added build badge with coverage report

### LINKED ISSUE #482: Add build badges and update test coverage config
What should be done:
1. Add CI build badge at the top of the https://github.com/ubiquity/ubiquity-dollar/blob/development/README.md with the current CI build status (success or fail)
2. Add test coverage badge with % of lines covered by tests at the top of the https://github.com/ubiquity/ubiquity-dollar/blob/development/README.md
3. Remove the following files from the test coverage report:

- `src/dollar/libs/*`
- `src/dollar/mocks/*`
- `src/dollar/utils/*`
- `src/manager/libraries/*`
- `src/ubiquistick/MockUBQmanager.sol`
- `test/*`

Build badge docs: https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/adding-a-workflow-status-badge

### PR DESCRIPTION
**Pull request by FibrinLab**
Resolves #482

### CODE REVIEW

File: .github/workflows/test-coverage-action.yml [hunk 1]
```diff
@@ -39,7 +39,7 @@ jobs:
         run: yarn workspace @ubiquity/contracts forge install
 
       - name: Run Forge tests
-        run: yarn workspace @ubiquity/contracts forge test --fork-url https://cloudflare-eth.com
+        run: yarn workspace @ubiquity/contracts forge test --fork-url https://eth-mainnet.g.alchemy.com/v2/WRRZczVcOh7GrDm7TeE7PwfVayvnXWe7
 
       - name: Coverage
-        run: yarn workspace @ubiquity/contracts forge coverage --fork-url https://cloudflare-eth.com
\ No newline at end of file
+        run: yarn workspace @ubiquity/contracts forge coverage --fork-url https://eth-mainnet.g.alchemy.com/v2/WRRZczVcOh7GrDm7TeE7PwfVayvnXWe7
```
Comment (0x4007) [inline .github/workflows/test-coverage-action.yml]:
```suggestion
        run: yarn workspace @ubiquity/contracts forge coverage --fork-url https://eth.ubq.fi/v1/mainnet
```

File: .github/workflows/test-coverage-action.yml [hunk 2]
```diff
@@ -39,7 +39,7 @@ jobs:
         run: yarn workspace @ubiquity/contracts forge install
 
       - name: Run Forge tests
-        run: yarn workspace @ubiquity/contracts forge test --fork-url https://cloudflare-eth.com
+        run: yarn workspace @ubiquity/contracts forge test --fork-url https://eth-mainnet.g.alchemy.com/v2/WRRZczVcOh7GrDm7TeE7PwfVayvnXWe7
```
Comment (0x4007) [inline .github/workflows/test-coverage-action.yml]:
```suggestion
        run: yarn workspace @ubiquity/contracts forge test --fork-url https://eth.ubq.fi/v1/mainnet
```

**Comment by 0x4007** [comment]
ChatGPT made this script to help conform the messages to conventional commits. Will need adjustments but it might help.

```typescript
import { spawn } from 'child_process';

const runCommand = (command: string, args: string[]): Promise<number> => {
  return new Promise((resolve, reject) => {
    const child = spawn(command, args);
    child.stdout.on('data', (data) => {
      console.log(`stdout: ${data}`);
    });
    child.stderr.on('data', (data) => {
      console.error(`stderr: ${data}`);
      reject(data);
    });
    child.on('close', (code) => {
      resolve(code);
    });
  });
};

const updateCommitMessage = async (hash: string, message: string) => {
  await runCommand('git', ['commit', '--amend', '-m', message, '--no-edit']);
  await runCommand('git', ['push', '--force-with-lease']);
};

const main = async () => {
  const log = await runCommand('git', ['log', '--pretty=format:"%h %s"']);
  const logLines = log.split('\n');
  for (const line of logLines) {
    const [hash, message] = line.split(' ');
    if (!message.startsWith('type(scope): description')) {
      const updatedMessage = `fix(${hash}): ${message}`;
      await updateCommitMessage(hash, updatedMessage);
    }
  }
};

main().catch((error) => {
  console.error(error);
});

```

**Comment by FibrinLab** [comment]
> ChatGPT made this script to help conform the messages to conventional commits. Will need adjustments but it might help.
> 
> ```ts
> import { spawn } from 'child_process';
> 
> const runCommand = (command: string, args: string[]): Promise<number> => {
>   return new Promise((resolve, reject) => {
>     const child = spawn(command, args);
>     child.stdout.on('data', (data) => {
>       console.log(`stdout: ${data}`);
>     });
>     child.stderr.on('data', (data) => {
>       console.error(`stderr: ${data}`);
>       reject(data);
>     });
>     child.on('close', (code) => {
>       resolve(code);
>     });
>   });
> };
> 
> const updateCommitMessage = async (hash: string, message: string) => {
>   await runCommand('git', ['commit', '--amend', '-m', message, '--no-edit']);
>   await runCommand('git', ['push', '--force-with-lease']);
> };
> 
> const main = async () => {
>   const log = await runCommand('git', ['log', '--pretty=format:"%h %s"']);
>   const logLines = log.split('\n');
>   for (const line of logLines) {
>     const [hash, message] = line.split(' ');
>     if (!message.startsWith('type(scope): description')) {
>       const updatedMessage = `fix(${hash}): ${message}`;
>       await updateCommitMessage(hash, updatedMessage);
>     }
>   }
> };
> 
> main().catch((error) => {
>   console.error(error);
> });
> ```

Thanks, will work on this.

**Comment by FibrinLab** [comment]
Ready for a review. Badges added to readme. I created a github workflow file to handle this

**Comment by 0x4007** [comment]
Your conventional commits aren't passing, thats why I gave you the script. I'll review when it passes. To be honest I'm not even sure how that happened (unless you intentionally disabled checks) if this is checked in the pre-commit hook.

I'll let the Slither tests check failure slide because we're in the middle of merging/fixing that now.

**Pull request closed without merging**